### DISCUSSION TITLE
Kill all running Progresses

### DISCUSSION THREAD

**Discussion by nicofirst1** [Textualize/rich#942] [Q&A] [answered]
I'm currently using a Progress bar to keep track of a training model.
My problem is that the training can sometimes raise a RuntimeError which is caught way up in the stack where I have no access to the instantiated Progress bar.
After catching this error the program reloads everything and re-instantiate another progress bar which starts immediately. The problem is that the old progress bar does not get deleted and still runs in the background conflicting with the new one. 
Is there a way to kill all the running progress bars without having a direct link to them?

**Comment by willmcgugan** [comment]
Not as such, but the Progress context manager should shut down threads on exit. Unless the RuntimeError is being handled within the progress context? Maybe create a shared Progress instance rather than create a new one each time.

**Comment by nicofirst1** [comment]
Unfortunately I'm using the Progress bar without a context manager (related to [this issue](https://github.com/willmcgugan/rich/issues/481) and [this discussion](https://github.com/willmcgugan/rich/discussions/482)).
But I found that I can access the progress thread using the threading module as below:
```
  for thread in threading.enumerate():
                if isinstance(thread, _RefreshThread):
                   kill progress
```
Is there an elegant way to kill the progress bar in this way?

**Reply by nicofirst1** [reply]
I tried 
```
 thread.done.set()
 thread.progress.disable = True
```
But it's not working

**Comment by willmcgugan** [comment]
I wouldn't recommend that approach.

Could you keep a reference to the Progress instance in a classvar? If one already exist, call `stop()` on it, before creating a new instance?

**Comment by nicofirst1** [comment, marked as answer]
I cannot keep a reference but i found that using the stop() method the old progress bar disappears.
```
  for thread in threading.enumerate():
                if isinstance(thread, _RefreshThread):
                    thread.progress.stop()
```
